webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DataStructures.scala

```diff
@@ -579,58 +579,114 @@ final case class BoundPolicyDraft(
    * for the targeted agent.
    * This is the place where we can check for variable consistency too, since we have the final
    * techniques and expanded variables (like: check that a mandatory variable was expanded to "").
+   *
+   * It's also the place where we add missing variables that are well defined, and where we check for
+   * constrains.
+   *
+   * If there is variable defined here but not in the technique, they are removed.
    */
   def toPolicy(agent: AgentType): Either[String, Policy] = {
     PolicyTechnique.forAgent(technique, agent).flatMap { pt =>
```

Comment:
this is the main change. Now we check for all variable here, and not only the one which were set.